soc/nordic/common/CMakeLists.txt

```diff
@@ -7,6 +7,8 @@ if(CONFIG_ARM AND CONFIG_SOC_FAMILY_NORDIC_NRF)
   zephyr_library_sources(soc_nrf_common.S)
 endif()
 
+zephyr_library_sources(dmm.c)
```

Comment:
nit: I guess some `IFDEF` is required for this line to avoid compiling dmm.c for targets without data cache

Reply:
sure, I will do it properly in final implementation for legacy devices like nRF52